### DISCUSSION TITLE
Dark Mode switch does not work properly in header

### DISCUSSION THREAD

**Discussion by laserir** [zauberzeug/nicegui#1830] [Q&A] [answered]
### Question

I discovered that a switch bound to ui.dark_mode() will not work properly (**with tab changes**) when included in ui.header(). Any idea why that is? Here is an example:
```py
from nicegui import app, ui

ui.dark_mode(True)

with ui.header(elevated=True).classes('glossy justify-between',replace='row items-center'):
	with ui.tabs() as tabs:
		ui.tab('Tab A')
		ui.tab('Tab B')
	ui.switch('Dark mode').bind_value(ui.dark_mode())   # Does NOT work here

with ui.tab_panels(tabs, value='Tab A'):
	with ui.tab_panel('Tab A'):
		ui.label('Content A')

	with ui.tab_panel('Tab B'):
		ui.label('Content B')
	
ui.switch('Dark mode').bind_value(ui.dark_mode())	# Works here

ui.run()
```
PS: In my code I can't set dark in ui.run(), as the mode is loaded depending on user after login.

**Comment by laserir** [comment]
PPS: Additionally... Is there a way to have the header change colors as well when in dark-mode?

**Reply by falkoschindler** [reply]
Yes, you can add Tailwind classes depending on the current mode:
```py
with ui.header().classes('dark:bg-gray-800'):
    ...
```

See https://tailwindcss.com/docs/dark-mode for more information.

**Reply by laserir** [reply]
Ah very nice! Had 'bg-dark' as class (dont know why) which is not part of the tailwind definition. Got the dark mode with different colors working as suggested:
`ui.header().classes('bg-slate-400 dark:bg-black')`

**Comment by falkoschindler** [comment, marked as answer]
I think the problem is that you have multiple `ui.dark_mode` elements interfering with each other. Their intended use would be something like this:
```py
dark_mode = ui.dark_mode(True)

with ui.header():
    with ui.tabs() as tabs:
        ui.tab('Tab A')
        ui.tab('Tab B')
    ui.switch('Dark mode').bind_value(dark_mode)

with ui.tab_panels(tabs, value='Tab A'):
    with ui.tab_panel('Tab A'):
        ui.label('Content A')
    with ui.tab_panel('Tab B'):
        ui.label('Content B')

ui.switch('Dark mode').bind_value(dark_mode)
```

**Reply by laserir** [reply]
Did the trick! Thanks a lot!
